enarx-prstate

```diff
@@ -25,19 +25,65 @@ def handle(issue, pr, responsible):
     if needed:
         pr.add_to_assignees(*needed)
 
-# If there are any changes requested, the author is responsible.
-for issue in github.search_issues(f"org:enarx is:pr is:public is:open -is:draft review:changes_requested"):
+# Look through all PRs that have a changes requested review present.
+for issue in github.search_issues(f"org:enarx is:pr is:public is:open -is:draft"):
+    parent_repo = issue.repository
+
     pr = issue.as_pull_request()
 
-    handle(issue, pr, {pr.user.login})
+    # Get the latest actionable reviews for all PR reviewers.
+    latest_reviews = {r.user.id: r for r in pr.get_reviews() if r.state == "CHANGES_REQUESTED" or r.state == "APPROVED"}
+    # Then, split that set into Approving Reviewers and Changes Requested reviews.
+    approving_reviewers = {u.user.login for (k, u) in latest_reviews.items() if u.state == "APPROVED"}
+    changes_requested_reviews = {u.user.id: u for (k, u) in latest_reviews.items() if u.state == "CHANGES_REQUESTED"}
 
-# Otherwise, requested reviewers and reviewers with dismissed reviews are.
-# This handles both where the author requests re-review and where, after
-# approvals, reviews were dismissed due to a new push.
-for issue in github.search_issues(f"org:enarx is:pr is:public is:open -is:draft -review:changes_requested"):
-    pr = issue.as_pull_request()
+    # Collect reviewers whose Changes Requested reviews have been addressed in a set.
+    addressed_reviewers = set()
+    for event in pr.get_issue_events():
+        if (event.requested_reviewer is not None and
+            event.requested_reviewer.id in changes_requested_reviews and
+            event.created_at > changes_requested_reviews[event.requested_reviewer.id].submitted_at):
+            # If the above condition is true, that means the PR author has
+            # re-requested a review from the reviewer after the reviewer's last
+            # CHANGES_REQUESTED review was submitted. That implies the PR is
+            # ready for re-review from that reviewer.
+            addressed_reviewers.add(event.requested_reviewer.login)
 
+    # Get all reviewers whose reviews were dismissed. They will need to be
+    # assigned regardless.
     dismissed = {r.user.login for r in pr.get_reviews() if r.state == 'DISMISSED'}
-    requested = {r.login for r in pr.get_review_requests()[0]}
-
-    handle(issue, pr, dismissed | requested)
+    
+    # Figure out how many unaddressed reviews there are.
+    unaddressed_reviews = len(changes_requested_reviews) - len(addressed_reviewers)
+    
+    # Get the minimum number of required approving reviews for the PR to merge.
+    required_reviews = parent_repo.get_branch(parent_repo.default_branch).get_required_pull_request_reviews().required_approving_review_count
```

Comment:
As central as this is to good, clean logic below, there is a caveat.

Apparently, even reading branch protection status requires that the user doing it is an admin on the repo. So, if we wanted this to work, `enarxbot` would need to be admin on every Enarx repo.